libs/sdk-backend-bear/src/backend/workspace/elements/elements.ts

```diff
@@ -100,4 +130,71 @@ class BearWorkspaceElementsQuery implements IElementQuery {
                 : () => Promise.resolve(emptyResult),
         };
     }
+
+    private createFilterExpressionFromAttributeFilters(attributeFilters: IElementQueryAttributeFilter[]) {
+        interface IFiltersGroupedByUri {
+            [overAttributeUri: string]: IElementQueryAttributeFilter[];
+        }
+
+        const groupsByConnectingAttribute = attributeFilters.reduce<IFiltersGroupedByUri>(
+            (groups, filter) => {
+                const { overAttribute } = filter;
+                if (!isUriRef(overAttribute)) {
+                    throw new NotSupported("Identifier refs are not supported in elements attribute filter");
+                }
+                groups[overAttribute.uri] = groups[overAttribute.uri]
+                    ? groups[overAttribute.uri].concat(filter)
+                    : [filter];
+                return groups;
+            },
+            {},
+        );
+
+        const groupedExpressionsByOverUri = Object.keys(groupsByConnectingAttribute).map((overAttribute) => {
+            const attributesAndElements = groupsByConnectingAttribute[overAttribute]
+                .map((parentFilter) => {
+                    const negativeFilter = isNegativeAttributeFilter(parentFilter.attributeFilter);
+                    const attributeElements = filterAttributeElements(parentFilter.attributeFilter);
+                    const displayForm = filterObjRef(parentFilter.attributeFilter);
+
+                    if (!isAttributeElementsByRef(attributeElements)) {
+                        throw new NotSupported(
+                            "Only attribute elements by ref are supported in elements attribute filter",
+                        );
+                    }
+                    if (!isUriRef(displayForm)) {
+                        throw new NotSupported(
+                            "Identifier refs are not supported in elements attribute filter",
+                        );
+                    }
+
+                    const elementsString = attributeElements.uris.map((uri) => `[${uri}]`).join(", ");
+                    const operatorString = negativeFilter ? "NOT IN" : "IN";
+                    return `[${displayForm.uri}] ${operatorString} (${elementsString})`;
+                })
+                .join(" AND ");
+
+            return `((${attributesAndElements}) OVER [${overAttribute}] TO [${this.attributeRef?.uri}])`;
+        });
+
+        return groupedExpressionsByOverUri.join(" AND ");
+    }
+
+    private createAfmForAttributeFilters(
+        attributeFilters: IElementQueryAttributeFilter[],
+    ): GdcExecuteAFM.IAfm {
+        if (!isUriRef(this.displayFormRef)) {
+            throw new NotSupported("Identifier refs are not supported in elements attribute filter");
```

Comment:
Why are they not supported? You can convert idRefs to uriRefs if needed with `objRefToUri`.

Reply:
In this case it's easy to fix but in the other cases where the refs are in array elements then it is harder to implement. The question is whether it's worth doing it when we know that on Bear these parameters will be always URI refs